### ISSUE TITLE
Feature Requests: Text Replace, Search Case Sensitivity Options, and Open Folder

### ISSUE THREAD

**Issue by QuinStudios** [RealRTTV/nbtworkbench#14]
I'll preface this by saying it's possible some of this functionality is implemented already, but I couldn't find it/figure it out from the main page, tool options, or version history. I also can't figure out how to actually view the hits returned from a search. ~I get the tooltip, but nothing changes and there are no visual indications about where the search results are~.

That said, these are the core features I find lacking and that prevent me from using the tool for my NBT editing needs:

**Open Folder:**
This one is the least critical, but is a quality of life feature that both NBTExplorer and NBT Studio have, which this tool doesn't seem to. Both those tools allow the user to open a folder instead of just an individual file, and then the individual compatible files (like region files) can be interacted with or searched together within a single tab. This is incredibly useful for large world files if you're searching for something specific, but don't know which region file it's contained within, since it allows you to just open the whole world and search all of its files.

**Search Case Sensitivity Options:**
The search option is case sensitive, and case sensitivity seemingly cannot be disabled. The ability to search in a non case-sensitive way is another feature that NBT Studio has, but this doesn't, and it makes it difficult to search words when you aren't positive about their capitalization. For instance, I'm trying to anonymize a world by replacing instances of real life names, but it's possible that letters of names were accidentally capitalized or un-capitalized in some instances, and to comprehensively find all of those potential instances is virtually impossible when there's case sensitivity.

**Search and Replace**
This is a feature that I don't think any NBT editors have, but would be the biggest life/time saver and QoL feature. Just a simple search and replace option/interface, to find any instances of given text and replace them with desired replacement text, not unlike most code editors. Again, in my instance I'm trying to anonymize a world, so for instance, the idea is that I'd be able to run a search for all instances of "Bob" and replace them with "REDACTED".

**Comment by RealRTTV** [comment]
just a fair warning, open folder is a *lot* of work to put in

**Comment by QuinStudios** [comment]
Yeah, that doesn't surprise me at all, hence why I mentioned it's the least critical. It's really great for quality of life and saving some time, and I'd love to see it added at some point, but not having it doesn't prevent functionality like the other two suggestions.

**Comment by RealRTTV** [comment]
nbtworkbench has a command line mode that allows searching directories

**Comment by QuinStudios** [comment]
How do I use the command line mode? I don't see any documentation about it, and I'm not familiar enough with general command line commands, if it's supposed to be intuitive to frequent command line users.

**Comment by RealRTTV** [comment]
yeah um... there are no docs on the command-line mode, you can do `nbtworkbench -?` to get a list of what you can do with the command line mode, and that's pretty much the docs for it

**Comment by RealRTTV** [comment]
2/3 are done, gonna push later tonight.

**Comment by RealRTTV** [comment]
for future reference, please make 3 separate issues, i need the dopamine from closing issues

**Comment by QuinStudios** [comment]
Will do. I assume Open Folder is the one that isn't done?

**Comment by RealRTTV** [comment]
I've added the other two, and many surrounding features for them. I would like to highlight that the intended way to search through file trees is using either tabs on important files, or when there's too many, using the cli mode of `nbtworkbench` (see README.md) to search through them efficiently
